virtualizarr/manifests/group.py

```diff
@@ -0,0 +1,36 @@
+from typing import TypeAlias
+
+from zarr.core.group import GroupMetadata
+
+from virtualizarr.manifests import ManifestArray
+
+ManifestDict: TypeAlias = dict[str, ManifestArray]
+
+
+class ManifestGroup:
+    """
+    Virtualized representation of multiple ManifestArrays as a Zarr Group.
+    """
+
+    _manifest_dict: ManifestDict
```

Comment:
instead of wrapping a dict, couldn't the `ManifestGroup` _be_ dict (or at least be a `Mapping`). That would allow one fewer layers of `._manifest` later

Reply:
The manifest group class allows adding group level attributes in the group metadata (e.g., https://github.com/zarr-developers/VirtualiZarr/blob/5231800d94ab120c5bbf75c6a6f520993d63e2a3/virtualizarr/tests/test_manifests/test_store.py#L55-L57), which would map to dataset / datatree node attrs in the xarray model. IIUC the tradeoff for one less `._manifest` later would be less readable code if we have a `_zarr_group_metadata` key or something like that, or is there a different way to support the group level attributes?

Reply:
I added a TODO for this idea in [`6bfdf4d` (#490)](https://github.com/zarr-developers/VirtualiZarr/pull/490/commits/6bfdf4d4a67b6f9860316f365fc5c986ce256067)